assets/dns/daemonset.yaml

```diff
@@ -89,18 +89,23 @@ spec:
           while true; do
             declare -A svc_ips
             for svc in "${services[@]}"; do
-              # Fetch service IP from cluster dns if present
-              ips=($(dig -t A @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"))
-
-              if [[ "$?" -eq 0 && "${#ips[@]}" -ne 0 ]]; then
-                svc_ips["${svc}"]="${ips[@]}"
-              else
-                # Try IPv6
-                ips=($(dig -t AAAA @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"))
+              # Fetch service IP from cluster dns if present. We make several tries
+              # to do it: IPv4, IPv6, IPv4 over TCP and IPv6 over TCP. The two last ones
+              # are for deployments with Kuryr on older OpenStack (OSP13) - those do not
+              # support UDP loadbalancers and require reaching DNS through TCP.
+              cmds=('dig -t A @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"'
+                    'dig -t AAAA @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"'
+                    'dig -t A +tcp @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"'
+                    'dig -t AAAA +tcp @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"')
+              declare -p cmds
+              for i in ${!cmds[*]}
```

Comment:
Why not just
```bash
for cmd in "${cmds[@]}"
```
and use the command directly?

Reply:
Right, I had some issues with that, but probably due to wrong escaping.

Reply:
Ah no, doing it the way you specify I'm getting elements of the array split by spaces. It needs to be like this.

```
[mdulko@dulek-workstation ~]$ for svc in "${services[@]}"; do
>   # Fetch service IP from cluster dns if present. We make several tries
>   # to do it: IPv4, IPv6, IPv4 over TCP and IPv6 over TCP. The two last ones
>   # are for deployments with Kuryr on older OpenStack (OSP13) - those do not
>   # support UDP loadbalancers and require reaching DNS through TCP.
>   cmds=('dig -t A @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"'
>         'dig -t AAAA @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"'
>         'dig -t A +tcp @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"'
>         'dig -t AAAA +tcp @"${NAMESERVER}" +short "${svc}.${CLUSTER_DOMAIN}"')
>   declare -p cmds
>   for cmd in ${cmds[@]}
>   do
>     echo $cmd
>   done
> done
declare -a cmds=([0]="dig -t A @\"\${NAMESERVER}\" +short \"\${svc}.\${CLUSTER_DOMAIN}\"" [1]="dig -t AAAA @\"\${NAMESERVER}\" +short \"\${svc}.\${CLUSTER_DOMAIN}\"" [2]="dig -t A +tcp @\"\${NAMESERVER}\" +short \"\${svc}.\${CLUSTER_DOMAIN}\"" [3]="dig -t AAAA +tcp @\"\${NAMESERVER}\" +short \"\${svc}.\${CLUSTER_DOMAIN}\"")
dig
-t
A
@"${NAMESERVER}"
+short
"${svc}.${CLUSTER_DOMAIN}"
dig
-t
AAAA
@"${NAMESERVER}"
+short
"${svc}.${CLUSTER_DOMAIN}"
dig
-t
A
+tcp
@"${NAMESERVER}"
+short
"${svc}.${CLUSTER_DOMAIN}"
dig
-t
AAAA
+tcp
@"${NAMESERVER}"
+short
"${svc}.${CLUSTER_DOMAIN}"

```

Reply:
I wonder if it would have taken less time to introduce a new command to the Go binary...

Reply:
Ack, although I bet, there is a better way for doing this.